src/DynamoCore/Graph/Nodes/NodeModel.cs

```diff
@@ -2307,52 +2306,56 @@ internal void RequestValueUpdate(EngineController engine)
         private void OnRenderPackageUpdateCompleted(AsyncTask asyncTask)
         {
             var task = asyncTask as UpdateRenderPackageAsyncTask;
-            var packages = new List<IRenderPackage>();
-            if (task.RenderPackages.Any())
-            {
-                packages.AddRange(task.RenderPackages);
-                packages.AddRange(OnRequestRenderPackages());
+            var packages = new RenderPackageCache();
 
+            if (!task.RenderPackages.IsEmpty())
+            {
+                packages.Add(task.RenderPackages);
+                packages.Add(OnRequestRenderPackages());
             }
-            OnRenderPackagesUpdated(packages);
 
+            OnRenderPackagesUpdated(packages);
         }
 
         /// <summary>
         ///
         /// </summary>
-        public event Func<IEnumerable<IRenderPackage>> RequestRenderPackages;
+        public event Func<RenderPackageCache> RequestRenderPackages;
```

Comment:
this and other public changes in this PR, should be documented in the API changes doc, and please investigate if this breaks revit or reach builds? There is a `RevitWatch3dViewModel` so I believe it might need to be updated as well.